### PR TITLE
New resource: mysql_global_variable

### PR DESCRIPTION
**Pull request by borissavelev**
To have state for MySQL global configuration with IaC I'd like to introduce new resource: mysql_global_variable

### CODE REVIEW

**Review by petoju — CHANGES_REQUESTED** [review]
My take on this:
1. I wouldn't use it as the changes are not persisted and MySQL restart removes them. That said, it could still be used for some ad-hoc settings. As it has a chance to be used, I'm not explicitly against merging it.
2. Please write tests. It's really easy for terraform - you can inspire yourself with mysql/resource_database_test.go. Something setting a (new) variable, changing it and "destroying it" (setting the default) would do it. I'm asking for this to be sure it works on all supported databases and there won't be critical issues I did not notice.
3. Consider writing a documentation inside website subfolder.

To sum up, I have some notes about the code and I'd like to see a test. Once that is done and there are no new issues, I'm happy to merge your changes.

**Comment by borissavelev** [comment]
Ah, good point about persistence. I'm using this with [tidb](https://docs.pingcap.com/tidb/dev/sql-statement-set-variable) and all global variables are persistent there by default.

For mysql 8 there is an option to use `persistent` keyword but I need to test it.

Thank you! Fixes and tests are coming soon)

**Comment by petoju** [comment]
Thanks for improving it and tests.

Merged.
